sql/backfill: TestVectorIndexMergingDuringBackfillWithPrefix failed

sql/backfill.TestVectorIndexMergingDuringBackfillWithPrefix [failed](https://mesolite.cluster.engflow.com/invocations/default/c201638e-00c2-43f0-b524-54cc8139bece?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2JhY2tmaWxsOmJhY2tmaWxsX3Rlc3Q=) on release-25.4 @ [095068c58091c9152c33c223af10ca71165c7ddd](https://github.com/cockroachdb/cockroach/commits/095068c58091c9152c33c223af10ca71165c7ddd):


```
=== RUN   TestVectorIndexMergingDuringBackfillWithPrefix
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestVectorIndexMergingDuringBackfillWithPrefix3437955577
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    backfill_test.go:409: 
        	Error Trace:	pkg/sql/backfill/backfill_test.go:409
        	Error:      	Received unexpected error:
        	            	dial tcp 127.0.0.1:45779: connect: connection refused
        	Test:       	TestVectorIndexMergingDuringBackfillWithPrefix
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestVectorIndexMergingDuringBackfillWithPrefix3437955577
--- FAIL: TestVectorIndexMergingDuringBackfillWithPrefix (830.59s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #156994 sql/backfill: TestVectorIndexMergingDuringBackfillWithPrefix failed [C-test-failure O-robot T-sql-queries branch-master release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestVectorIndexMergingDuringBackfillWithPrefix.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-56338

--- a/pkg/sql/backfill/backfill_test.go
+++ b/pkg/sql/backfill/backfill_test.go
@@ -346,6 +346,13 @@ func TestVectorIndexMergingDuringBackfillWithPrefix(t *testing.T) {
 	blockBackfill := make(chan struct{})
 	backfillBlocked := make(chan struct{})
 
+	var testTenant base.DefaultTestTenantOptions
+	if syncutil.DeadlockEnabled {
+		// The cluster can get overloaded under deadlock with external process
+		// mode.
+		testTenant = base.TestSkipForExternalProcessMode()
+	}
+
 	ctx := context.Background()
 	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{
 		Knobs: base.TestingKnobs{
@@ -357,6 +364,7 @@ func TestVectorIndexMergingDuringBackfillWithPrefix(t *testing.T) {
 				},
 			},
 		},
+		DefaultTestTenant: testTenant,
 	})
 	defer srv.Stopper().Stop(ctx)
 	sqlDB := sqlutils.MakeSQLRunner(db)

